kong/plugins/aws-lambda/handler.lua

```diff
@@ -32,11 +35,34 @@ end
 
 
 local function initialize()
+  LAMBDA_SERVICE_CACHE = lrucache.new(1000)
   AWS_GLOBAL_CONFIG = aws_config.global
   AWS = aws()
   initialize = nil
 end
 
+local build_cache_key do
+  -- Use AWS Service related config fields to build cache key
+  -- so that service object can be reused between plugins and
+  -- vault refresh can take effect when key/secret is rotated
+  local SERVICE_RELATED_FIELD = { "timeout", "keepalive", "aws_key", "aws_secret",
+                                  "aws_assume_role_arn", "aws_role_session_name",
+                                  "aws_region", "host", "port", "disable_https",
+                                  "proxy_url", "aws_imds_protocol_version" }
```

Comment:
since `SERVICE_RELATED_FIELD` is a fixed table, what are the benefits of placing it in a closure?

Reply:
The purpose is to show that this fixed table is only involved in this closure, and not being used by any other code. No benefits other than that

Reply:
Does placing it at the module level avoid allocating a table for each closure during each run?

Reply:
I think it only runs once, no? Since the table definition is in do-end block, not the function definition

Reply:
> I think it only runs once, no? Since the table definition is in do-end block, not the function definition

right, my mistake.